bundler/lib/bundler/definition.rb

```diff
@@ -385,8 +385,8 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
 
       # Check if it is possible that the source is only changed thing
       if (new_deps.empty? && deleted_deps.empty?) && (!new_sources.empty? && !deleted_sources.empty?)
-        new_sources.reject! {|source| (source.path? && source.path.exist?) || equivalent_rubygems_remotes?(source) }
-        deleted_sources.reject! {|source| (source.path? && source.path.exist?) || equivalent_rubygems_remotes?(source) }
+        new_sources.reject! {|source| source.path? && source.path.exist? }
+        deleted_sources.reject! {|source| source.path? && source.path.exist? }
```

Comment:
\nit Would it make sense to add a dedicated method for the `source.path? && source.path.exist?` check on source? IMO this would make the code a bit more readable.

Reply:
Yes, I actually had some more refactorings on gemfile vs lockfile equivalence, and I was planning on removing this whole block of code, since I don't think it's needed. But I decided to leave them out of this PR for now to keep it smaller.

Reply:
I created #5120 that completely removes this block of code.